tests/test_autonomy/test_cli/test_scaffold_fsm.py

```diff
@@ -65,25 +54,39 @@ def setup_class(cls) -> None:
         cls.create_agents(cls.agent_name, is_local=cls.IS_LOCAL, is_empty=cls.IS_EMPTY)
         shutil.move(str(cls.t / cls.agent_name), str(cls.t / "packages"))
 
-    def test_run(self, fsm_spec_file: Path) -> None:
-        """Test run."""
+    @pytest.mark.order(1)
+    def test_scaffold_fsm(self, fsm_spec_file: Path) -> None:
+        """Test scaffold fsm."""
 
-        my_skill = f"test_{fsm_spec_file.parts[-2]}"
-        self.cli_options[-3] = my_skill
         path_to_spec_file = Path(ROOT_DIR) / fsm_spec_file
-        args = [*self.cli_options, path_to_spec_file]
+        packages_path = str(Path(ROOT_DIR) / Path(PACKAGES))
+        my_skill = f"{fsm_spec_file.parts[-2]}"
+        scaffold_args = ["scaffold", "fsm", my_skill, "--local", "--spec"]
+        args = ["--registry-path", packages_path, *scaffold_args, path_to_spec_file]
         result = self.run_cli_command(*args, cwd=self._get_cwd())
         assert result.exit_code == 0
 
+    @pytest.mark.order(2)
     def test_imports(
         self, fsm_spec_file: Path, monkeypatch: pytest.MonkeyPatch
     ) -> None:
         """Test imports of scaffolded modules"""
 
         monkeypatch.syspath_prepend(self.t)
         path = self.t / self.agent_name
-        for file in path.glob("**/*.py"):
+        for file in path.rglob("**/*.py"):
             module_spec = importlib.util.spec_from_file_location("name", file)
             assert isinstance(module_spec, ModuleSpec)
             module_type = importlib.util.module_from_spec(module_spec)
             module_spec.loader.exec_module(module_type)  # type: ignore
+
+    @pytest.mark.order(3)
+    def test_autonomy_test(self, fsm_spec_file: Path) -> None:
+        """Run autonomy test on the scaffolded skill"""
+
+        prefix = self.t / "packages" / self.agent_name / "skills"
+        path = prefix / fsm_spec_file.parent.parts[-1]
+        packages_dir = str(Path(self._get_cwd()).parent)
+        args = ["test", "by-path", str(path)]
+        result = self.run_cli_command(*args, cwd=packages_dir)
+        assert result.exit_code == 0
```

Comment:
please read PR description. This will fail unless we add a fingerprint ignore pattern for hidden files.

Reply:
The way I see it is there are several problems with this test setup, a test should depend on the other tests in the same class so please fix this by making sure that each test runs isolated and unaffected by other tests. 

Re the `.pytest` folder, it get's created in the same directory as you run the command from so If you're not running the command from the package folder itself then it should not be a problem